lib/Request_Ride_Loc.dart

```diff
@@ -62,62 +68,196 @@ class _RequestRideLocState extends State<RequestRideLoc> {
+                  )
                 ],
               ),
-              SizedBox(height: 40.0),
+              SizedBox(height: 20.0),
               Form(
+                key: _formKey,
                 child: Column(
                   children: <Widget>[
-                    _buildFromField(),
-                    SizedBox(height: 40.0),
-                    _buildToField(),
-                    SizedBox(height: 10.0)
+                    FromLocation(fromCtrl: fromCtrl),
+                    SizedBox(height: 10.0),
+                    ToLocation(toCtrl: toCtrl),
                   ],
                 ),
               ),
               Expanded(
                 child: Align(
-                  alignment: Alignment.bottomCenter,
+                    alignment: Alignment.bottomCenter,
                     child: Padding(
                       padding: const EdgeInsets.only(bottom: 20.0),
                       child: ButtonTheme(
                         minWidth: MediaQuery.of(context).size.width * 0.8,
                         height: 45.0,
-                        shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(3)),
+                        shape: RoundedRectangleBorder(
+                            borderRadius: BorderRadius.circular(3)),
                         child: RaisedButton(
-                          onPressed: (){
-                            Navigator.push(context, new MaterialPageRoute(builder: (context) => RequestRideTime()));
+                          onPressed: () {
+                            Navigator.push(
+                                context,
+                                new MaterialPageRoute(
+                                    builder: (context) =>
+                                        RequestRideTime(ride: widget.ride)));
+                            widget.ride.setFromLocation(fromCtrl.text);
+                            widget.ride.setToLocation(toCtrl.text);
                           },
                           elevation: 3.0,
                           color: Colors.black,
                           textColor: Colors.white,
                           child: Text('Next'),
                         ),
                       ),
-                    )
-                ),
+                    )),
               )
             ],
           ),
-        )
+        ));
+  }
+}
+
+class FromLocation extends StatefulWidget {
+  final TextEditingController fromCtrl;
+
+  FromLocation({Key key, this.fromCtrl}) : super(key: key);
+
+  @override
+  _FromLocationState createState() => _FromLocationState();
+}
+
+class _FromLocationState extends State<FromLocation> {
+  Widget _fromField(BuildContext context, List<Location> locations) {
+    return TypeAheadFormField(
+      textFieldConfiguration: TextFieldConfiguration(
+          controller: widget.fromCtrl,
+          decoration: InputDecoration(
+              labelText: 'From',
+              hintText: 'From',
```

Comment:
You can remove the hintText because the labelText acts as both the hint and the label--with the hintText, when you first select the text field, you have both the "From" at the top and the "From" in the box, but with just the labelText, you'll have the "From" in the text field before you focus on it, and then it'll move to the top when you focus on the text field. (this applies to all of the text fields on the flow)